tests/jballerina-unit-test/src/test/resources/test-src/expressions/let/let-expression-negative.bal

```diff
@@ -33,3 +33,26 @@ function wrongTypes() {
 function fun() returns int {
     return 1;
 }
+
+function invalidTypesWithAnonymousRecord() {
+    record {
+        byte i;
+        int j?;
+    } rec1 = let int v = 160 in {i: v};
+
+    record {
+        string i;
+        int j?;
+    } rec2 = let string v = "160" in {i: v};
+    
+    rec2.j = "invalid";
```

Comment:
Since `rec2`'s type will be the anonymous record type irrespective of the RHS expression, IMO as a let expression test it may be better to test an invalid expression for `j` in the mapping constructor expression itself.

```suggestion
    } rec2 = let string v = "160" in {i: v, j: "invalid"};
```